nix/package.nix

```diff
@@ -38,19 +42,44 @@ writeShellScriptBin "agenix" ''
     exit 1
   }
 
-  case "$1" in
-    "help"|"--help"|"-help"|"-h")
-      show_help
-      exit 1
-      ;;
-
-    ${lib.concatStringsSep "|" allApps})
-      APP=$1
-      shift
-      echo "Collecting information about hosts. This may take a while..."
-      exec nix run .#agenix-rekey.apps.${lib.escapeShellArg stdenv.hostPlatform.system}."$APP" -- "$@"
-      ;;
-
-    *) die "Unknown command: $1" ;;
-  esac
+  APP=""
+  SHOW_TRACE_ARG=""
+  # Various Bash versions treat empty arrays as unset, which then trigger
+  # unbound variable errors.
+  PASS_THRU_ARGS=("")
+  while [[ $# -gt 0 ]]; do
+    case "$1" in
+      "help"|"--help"|"-help"|"-h")
+        show_help
+        exit 1
+        ;;
+      "--show-trace")
+        # It is potentially desirable to use --show-trace in the subcommand as
+        # well as this command.  To do so, the --show-trace argument must be
+        # provided before (agenix) or after (subcommand) to indicate which one
+        # is to be used.  We account for this here.
+        if [[ $APP == "" ]]; then
+          SHOW_TRACE_ARG='--show-trace'
+        else
+          PASS_THRU_ARGS+='--show-trace'
```

Comment:
This must be in parenthesis `+=("--show-trace")` or otherwise it will directly concat to the first argument